File: src/main/kotlin/io/github/dqualizer/dqlang/types/dam/architecture/apischema/field/FieldItem.java
package io.github.dqualizer.dqlang.types.dam.architecture.apischema.field;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import java.util.Map;
import java.util.Set;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Data
public class FieldItem {

    private String path;
    @JsonProperty("operation_id")
    private String operationID;
    private String operation;
    private Set<Input> input;
    private Map<String, DataType> body;
    private Set<Map<String, Output>> output;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/dam/architecture/apischema/field/FieldItem.kt
package io.github.dqualizer.dqlang.types.dam.architecture.apischema.field

import com.fasterxml.jackson.annotation.JsonProperty

data class FieldItem(
    val path: String? = null,

    @JsonProperty("operation_id")
    val operationID: String? = null,

    val operation: String? = null,

    val input: Set<Input>? = HashSet(),

    val body: Map<String, DataType>? = HashMap(),

    val output: Set<Map<String, Output>>? = HashSet()
)
